crates/paimon/src/spec/types.rs

```diff
@@ -601,6 +765,57 @@ impl Default for LocalZonedTimestampType {
     }
 }
 
+impl FromStr for LocalZonedTimestampType {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if !s.starts_with(serde_utils::LocalZonedTimestamp::NAME) || !s.contains("WITH") {
+            return DataTypeInvalidSnafu {
+                message:
+                    "Invalid LocalZonedTimestamp type. Expected string to start with 'TIMESTAMP'.",
+            }
+            .fail();
+        }
+
+        let Some(open_bracket) = s.find('(') else {
+            return DataTypeInvalidSnafu {
+                message: "Invalid LocalZonedTimestamp specification. Missing opening bracket.",
+            }
+            .fail();
+        };
+        let Some(close_bracket) = s.find(')') else {
```

Comment:
What about extract a method to extract the `(number)` format ? There are almost 8 same usage